Editor/Windows/AssetPaletteWindowHeader.cs

```diff
@@ -207,22 +243,26 @@ private void SetSortModeAndSortCurrentEntries(SortModes sortMode)
 
         private void DoCollectionDropDown(Rect collectionRect)
         {
-            string[] existingCollectionGuids = AssetDatabase.FindAssets($"t:{typeof(AssetPaletteCollection).Name}");
+            
+            string[] existingCollectionGuids = AssetDatabase.FindAssets($"t:{nameof(AssetPaletteCollection)}");
 
             // Allow a new collection to be created.
             GenericMenu dropdownMenu = new GenericMenu();
 
             // Add any existing collections that we find as options.
-            foreach (string collectionGuid in existingCollectionGuids)
+            for (int i = 0; i < existingCollectionGuids.Length; i++)
             {
-                bool isCurrentCollection = collectionGuid == CurrentCollectionGuid;
+                string collectionGuid = existingCollectionGuids[i];
+                bool isCurrentCollection = string.Equals(
+                    collectionGuid, CurrentCollectionGuid, StringComparison.Ordinal);
                 string collectionName = Path.GetFileNameWithoutExtension(AssetDatabase.GUIDToAssetPath(collectionGuid));
-                dropdownMenu.AddItem(
-                    new GUIContent(collectionName), isCurrentCollection, LoadExistingCollection, collectionGuid);
+                dropdownMenu.AddItem(new GUIContent(collectionName), isCurrentCollection, LoadExistingCollection, collectionGuid);
             }
 
-            if (existingCollectionGuids.Length > 0)
-                dropdownMenu.AddSeparator("");
+            dropdownMenu.AddItem(new GUIContent("Personal Palette"), string.Equals(CurrentCollectionGuid, personalPaletteGuid, StringComparison.Ordinal), 
```

Comment:
Maybe the Personal Palette should be at the top if it's going to be the new default 🤔